lib/services/remote_sync_service.dart

```diff
@@ -214,12 +222,48 @@ class RemoteSyncService {
     }
   }
 
+  Future<void> _syncDeviceCollectionFilesForUpload() async {
+    final devicePathCollections =
+        await FilesDB.instance.getDevicePathCollections();
+    devicePathCollections.removeWhere((element) => !element.sync);
+    await _createCollectionsForDevicePath(devicePathCollections);
+  }
+
+  Future<void> _createCollectionsForDevicePath(
+    List<DevicePathCollection> devicePathCollections,
+  ) async {
+    for (var devicePathCollection in devicePathCollections) {
+      int deviceCollectionID = devicePathCollection.collectionID;
+      if (deviceCollectionID != -1) {
```

Comment:
Could it be `null`?

Reply:
No, the default value is -1 for devicePathCollection.collectionID and we are never setting it to null.